### PR TITLE
failing test for customer use case

### PR DESCRIPTION
**Pull request by JerryShea**

### CODE REVIEW

File: src/main/java/net/openhft/chronicle/wire/WireMarshaller.java [hunk 1]
```diff
@@ -2022,6 +2022,8 @@ else if (type == Set.class)
             componentType = extractClass(computeActualTypeArguments(Collection.class, field)[0]);
             if (componentType != Object.class) {
                 isLeaf = !Throwable.class.isAssignableFrom(componentType)
+                        // Don't recurse into the same class
+                        && !componentType.equals(field.getDeclaringClass())
```
Comment (david-ry4n) [inline src/main/java/net/openhft/chronicle/wire/WireMarshaller.java:2026]: Wasn't sure if this should be !componentType.isAssignmentFrom(field.getDeclaringClass())

**Comment by JerryShea** [comment]
I created the PR so can't approve, but I do! Please wait for @peter-lawrey to give his blessing.

We will need to backport to 25 and 24 - hopefully that is not too difficult

And we should create an issue for release notes

File: src/main/java/net/openhft/chronicle/wire/WireMarshaller.java (hunk 1 above)
Comment (JerryShea) [reply src/main/java/net/openhft/chronicle/wire/WireMarshaller.java:2026]: I think equality is correct as the test whereby the types were not the same but one assignable from the other, seemed to work OK

**Comment by david-ry4n** [comment]
This closes #915

**Review by peter-lawrey — APPROVED** [review]

**Pull request merged**